src/librustc/ty/context.rs

```diff
@@ -2139,10 +2138,9 @@ macro_rules! intern_method {
                 // See comments on Lift for why we can't use that.
                 if !($keep_in_local_tcx)(&v) {
                     if !self.is_global() {
-                        let v = unsafe {
-                            mem::transmute(v)
-                        };
-                        let i = ($alloc_to_ret)(self.global_interners.arena.$alloc_method(v));
+                        let i: &$lt_tcx $ty = $alloc_method(&self.global_interners.arena, v);
+                        // Cast to 'gcx
+                        let i = unsafe { mem::transmute(i) };
```

Comment:
That doesn't work since `$ty` uses `'tcx` lifetimes, hence the need for `transmute`.

Reply:
I didn't mean that the transmute should be removed, rather that the binding is explicitly annotated for more clarity:
```rust
let i: &$lt_tcx $ty = $alloc_method(&self.global_interners.arena, v);
// Cast to 'gcx
let i: &'gcx $ty = unsafe { mem::transmute(i) };
```

Reply:
I think @Zoxc's point is that `$ty` already has explicit lifetimes in it, so it wouldn't work.
OTOH, since this is a reference, maybe pointer casting should be used?

Reply:
I can probably just refactor the macros to get rid of the transmute here.